src/app/modules/data-set/annotation/annotation.component.ts

```diff
@@ -103,9 +93,9 @@ export class AnnotationComponent implements OnInit {
     this.addAnnotation(annotation);
   }
 
-  private initSmellsOrHeuristics(input: Map<string, string[]>, smellsOrHeuristics: Map<InstanceType, string[]>): void {
+  private initHeuristics(input: Map<string, string[]>, heuristics: Map<string, string[]>): void {
```

Comment:
You should think about introducing an object model on the front-end to avoid dictionaries. It is not necessary for this single case, but something to keep in mind.